onnxruntime/core/providers/openvino/backend_manager.cc

```diff
@@ -162,9 +168,19 @@ Status BackendManager::ExportCompiledBlobAsEPCtxNode(const onnxruntime::GraphVie
     model_blob_str = model_blob_stream.str();
     ORT_ENFORCE(model_blob_str.size() != 0);
   } else {
-    std::ofstream f(graph_name + ".blob", std::ios::out | std::ios::trunc | std::ios::binary);
-    compiled_model.export_model(f);
-    model_blob_str = graph_name + ".blob";
+    // Remove extension so we can append suffix to form the complete name of output graph
+    auto blob_name = [&]() {
+      size_t dot = graph_name.find_last_of(".");
+      if (dot == std::string::npos) return graph_name;
+      return graph_name.substr(0, dot);
+    }();
+    std::ofstream blob_file(blob_name + ".blob",
```

Comment:
If `ep.context_file_path` contains directories that do not exist, cache generation seems to fail, whereas the EP option `cache_dir` also supports such paths (creates intermediate directories automatically). It would be nice to make these two options consistent.

Reply:
Also, maybe worth returning an error if `blob_name + ".blob" == graph_name`, because in that case the generated ONNX file overwrites the blob file. It is never actually checked that the suffix of `graph_name` is `.onnx`, and probably there is no need to introduce such restriction anyway.

Reply:
@mnidza  The .blob file contains the compiled model whereas the generated ONNX file has reference to this .blob file. Can you elaborate on the scenario where it gets overwritten? 

Also are you suggesting us to add directories if it doesnt exist with ep.context_file_path option?

Reply:
> Can you elaborate on the scenario where it gets overwritten?

It gets overwriten if `ep.context_file_path` is set to `foo.blob` for example. It happened to me, because at first I thought that `ep.context_file_path` has the same interpretation as the EP option `cache_dir`.

> Also are you suggesting us to add directories if it doesnt exist with ep.context_file_path option?

Yes, because this is what happens with the EP option `cache_dir`.

In any case, none of these are super crucial, so please feel free to ignore if it's too much trouble.

Reply:
We are planning for a follow up PR by early next week. We will accomodate your review comments in the new PR as it requires validation from our teams. Thanks for the inputs @mnidza

Reply:
closing this comment , feedback to be addressed in follow-up PR